### PR TITLE
Add extra tests to the suite

### PR DESCRIPTION
**Pull request by PapaCharlie**
The follow tests were added:
- An action that echos bytes passed in, allows testing the JSON encoding for bytes
- A finder with a complex record as a parameter, allows for testing of URL encoding of records
- A finder with an optional parameter, allows for testing of optional parameters. The type of the
  optional parameter is also bytes, which can be used to test URL encoding of bytes
- A finder with a map as a parameter, allows for testing of map URL encoding and the special
  case of empty maps, or maps with empty strings
- A finder with an array as a parameter, allows for testing of array URL encoding and the special
  case of an array with an empty string
- A new test for a finder where one of the parameters is the empty string
- A new test for a finder where one of the parameters is the empty byte array

Also ensure JettyRunWar actually builds the server before running it, otherwise new
resources/methods would not show up.

### CODE REVIEW

File: client-testsuite/src/test/java/com/linkedin/pegasus/test/TestRestClientWithManualAssertions.java [hunk 1]
```diff
@@ -40,17 +43,20 @@
 {
   private final static String EXCEPTION_CLASS = "com.linkedin.restli.server.RestLiServiceException";
 
+  public static final String V1 = "v1";
+  public static final String V2 = "v2";
+
   public TestRestClientWithManualAssertions() throws Exception
   {
-    super(System.getProperty("test.projectDir"));
+    super("/Users/pchesnai/code/personal/go-restli/internal/tests/rest.li-test-suite/client-testsuite/");
```
Comment (nickibi) [inline client-testsuite/src/test/java/com/linkedin/pegasus/test/TestRestClientWithManualAssertions.java:51]: what's the reason to hard code it?

**Review by nickibi — COMMENTED** [review]
minor issue, otherwise LGTM.

File: client-testsuite/src/test/java/com/linkedin/pegasus/test/TestRestClientWithManualAssertions.java (hunk 1 above)
Comment (PapaCharlie) [reply client-testsuite/src/test/java/com/linkedin/pegasus/test/TestRestClientWithManualAssertions.java:51]: I think that was a mistake, I accidentally tracked this

**Comment by PapaCharlie** [comment]
A lot of these tests are actually redundant with other ones. I will spend some time revising this and I will open this PR once I have reduced the duplication

**Pull request closed without merging**